src/main/java/org/folio/orders/utils/HelperUtils.java

```diff
@@ -1097,4 +1097,88 @@ public static PoLine convertToPoLine(CompositePoLine compPoLine) {
             return null;
           });
   }
+
+  public static CompositePoLine convertToCompositePoLine(JsonObject poLine) {
+    poLine.remove(REPORTING_CODES);
+    poLine.remove(ALERTS);
+    return poLine.mapTo(CompositePoLine.class);
+  }
+
+  public static CompletableFuture<CompositePurchaseOrder> validateIfPOProtectedFieldsChanged(CompositePurchaseOrder compPO,
+      CompositePurchaseOrder compPOFromStorage) {
+    if (!compPOFromStorage.getWorkflowStatus()
+      .equals(CompositePurchaseOrder.WorkflowStatus.PENDING)) {
+      Set<String> fields = new HashSet<>();
+      for (String field : POProtectedFields.getFieldNames()) {
+        try {
+          if (isFieldChanged(compPO, compPOFromStorage, field)) {
+            fields.add(field);
+          }
+        } catch (IllegalAccessException e) {
+          throw new CompletionException(e);
+        }
+      }
+
+      verifyProtectedFieldsChanged(fields);
+    }
+
+    return completedFuture(compPOFromStorage);
+
+  }
+
+  public static Set<String> findChangedPoLineProtectedFields(CompositePoLine compPOLine, CompositePoLine compPOLineFromStorage) {
+    Set<String> fields = new HashSet<>();
+    for (String field : POLineProtectedFields.getFieldNames()) {
+      try {
+        if (field.contains(".")) {
```

Comment:
I'd like to suggest some changes, but it's up to you.
Instead of listing all the options with subobjects and then looking for and splitting them by a dot, I would add a list of "subobjects" field to POLineProtectedFields. This would allow you to get all the subobjects at once, like getSubobjects(), and if the list is not empty then check them in the loop.
I think it would simplify the code a bit.

Reply:
As an alternative option instead of reflection, `JsonObject` can be used to get value from by path. RMB provides [JsonPathParser](https://github.com/folio-org/raml-module-builder/blob/v26.0.0/domain-models-runtime/src/main/java/org/folio/rest/tools/parser/JsonPathParser.java) to simplify work with nested properties.

Reply:
Good find, Using JsonPathParser for parsing the json